grr/server/grr_response_server/gui/ui/store/client_page_facade.ts

```diff
@@ -247,19 +267,19 @@ export class ClientPageStore extends ComponentStore<ClientPageState> {
           );
 
   /** An effect fetching a client with a given id and updating the state. */
-  readonly selectClient = this.effect<string>(
+  readonly fetchClient = this.effect<void>(
       obs$ => obs$.pipe(
+          switchMapTo(this.select(state => state.clientId)),
```

Comment:
clientId might be null when somebody subscribes to `selectedClient` before calling `selectClient`. Please account for this and do not call `httpApiService.fetchClient`.

Reply:
I will filter undefined values of `clientId`